src/main/java/core/basesyntax/service/TransactionParser.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.Transaction;
+
+public class TransactionParser {
+    public Transaction parseLine(String line) {
+        String[] parts = line.split(",");
+        String operation = parts[0];
+        String fruitName = parts[1];
+        int quantity = Integer.parseInt(parts[2]);
+        return new Transaction(operation, fruitName, quantity);
+    }
```

Comment:
The `TransactionParser` class should be responsible solely for parsing and should not be aware of the structure of a `Transaction`. Consider creating a `TransactionDto` class to hold raw transaction data, which can then be validated and converted into a `Transaction` object by another service.